### PR TITLE
[ADD] account_due_list_days_overdue - refactorization

### PR DESCRIPTION
**Pull request by JordiBForgeFlow**
This is an experiment to try to create dynamic overdue periods...

cc @hbrunn

### CODE REVIEW

**Comment by hbrunn** [comment]
see https://github.com/Eficent/account-payment/pull/1

**Comment by JordiBForgeFlow** [comment]
@hbrunn Now it works when you create a new overdue term, but only when you restart the server. Not sure that calling _register_hook on the account.overdue.term in create, write, helps (I tried, with no success).

**Comment by hbrunn** [comment]
try calling https://github.com/OCA/OCB/blob/8.0/openerp/modules/registry.py#L486 instead of the register hook, this signals to all registries they have to reload, probably you're in a multi process environment where not all processes know about your change, and if you're lucky, this also calls the register hook again

**Comment by JordiBForgeFlow** [comment]
@hbrunn Now works like a charm ;) Thanks!!!
cc @elicoidal

File: account_due_list_days_overdue/README.rst [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+Usage
+=====
+
+To use this module, you need to go to:
+
+* Invoicing / Journal Entries / Payments and due list
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
```
Comment (elicoidal) [inline account_due_list_days_overdue/README.rst:36]: recommended:
:width: 600px

**Comment by elicoidal** [comment]
Good news :)
Will test on runbot when available

**Comment by JordiBForgeFlow** [comment]
@elicoidal @hbrunn Travis is green, with nearly 90% of test coverage :)

File: account_due_list_days_overdue/README.rst (hunk 1 above)
Comment (JordiBForgeFlow) [inline account_due_list_days_overdue/README.rst:36]: What do you mean by that?

**Comment by JordiBForgeFlow** [comment]
cc @jgrandguillaume This one is also worth a look.

**Comment by jgrandguillaume** [comment]
Hi,

Thanks for the work, I've just tested it on runbot and works great, even with multiple payment terms and payment.

:+1: 

Regards,

Joël

**Comment by JordiBForgeFlow** [comment]
Thanks @jgrandguillaume. I sweated blood on this one. Thankfully @hbrunn helped me a lot to resolve. A key concept that we now understand is how to dynamically add fields (including computed fields) to a any model.

File: account_due_list_days_overdue/README.rst (hunk 1 above)
Comment (elicoidal) [inline account_due_list_days_overdue/README.rst:36]: ignore

**Comment by elicoidal** [comment]
Tested :+1: 

2 small details about the menus:
- configuration entry should be in configuration/miscellaneous after payment terms
- report should be either in customer menu (more accessible) or in reporting/generic/partner (more logical)

**Comment by elicoidal** [comment]
another small nice to have: when group by in the reporting, the total should be displayed (like DR/CR/Residual)

**Comment by JordiBForgeFlow** [comment]
@elicoidal I don't think that it is possible to sum on group by in computed fields. And these fields cannot be stored because the value depends on current date.

I believe that this data model can be used to build a qweb report that shows the total due amounts by periods, grouped by partner.

**Comment by elicoidal** [comment]
That is what I suspected.

**Comment by eLBati** [comment]
Thanks :+1:

**Comment by eLBati** [comment]
@jbeficent what do you think about @elicoidal remarks at https://github.com/OCA/account-payment/pull/94#issuecomment-233118677 ?

**Comment by JordiBForgeFlow** [comment]
IMHO this resolves the need better: https://github.com/OCA/account-payment/pull/94

**Comment by eLBati** [comment]
@jbeficent you are linking this same PR. What did you mean?

**Pull request merged**